File: frontend/src/types/note.ts
export interface Note {
  id: string
  title: string
  content_markdown: string
  topic_id: string | null
  created_at: string
  updated_at: string
}

export interface NoteCreate {
  title: string
  content_markdown?: string
  topic_id?: string | null
}

export interface NoteUpdate {
  title?: string
  content_markdown?: string
  topic_id?: string | null
}

File: backend/app/schemas/note.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class NoteCreate(BaseModel):
    title: str = Field(min_length=1, max_length=255)
    content_markdown: str = ""
    topic_id: int | None = None


class NoteUpdate(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=255)
    content_markdown: str | None = None
    topic_id: int | None = None


class NoteRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    content_markdown: str
    topic_id: int | None
    created_at: datetime
    updated_at: datetime
